examples/experimental/dagster-airlift/dagster_airlift/core/sensor.py

```diff
@@ -43,83 +55,58 @@ def build_airflow_polling_sensor(
-                }
-                # Add dag materialization
-                dag_metadata = {
-                    **metadata,
-                    "Run Details": MarkdownMetadataValue(f"[View Run]({dag_run.url})"),
-                    "Start Date": TimestampMetadataValue(dag_run.start_date),
-                    "End Date": TimestampMetadataValue(dag_run.end_date),
-                }
-                materializations_to_report.append(
-                    (
-                        dag_run.end_date,
-                        AssetMaterialization(
-                            asset_key=dag_key,
-                            description=dag_run.note,
-                            metadata=dag_metadata,
-                        ),
-                    )
-                )
-                asset_check_keys_to_request.update(unmigrated_info.checks_per_key[dag_key])
-                task_runs = {}
-                for task_id, asset_key in task_keys:
-                    task_run: TaskInstance = task_runs.get(
-                        task_id, airflow_instance.get_task_instance(dag_id, task_id, dag_run.run_id)
-                    )
-                    task_runs[task_id] = task_run
-                    task_metadata = {
-                        **metadata,
-                        "Run Details": MarkdownMetadataValue(f"[View Run]({task_run.details_url})"),
-                        "Task Logs": MarkdownMetadataValue(f"[View Logs]({task_run.log_url})"),
-                        "Start Date": TimestampMetadataValue(task_run.start_date),
-                        "End Date": TimestampMetadataValue(task_run.end_date),
-                    }
-                    materializations_to_report.append(
-                        (
-                            task_run.end_date,
-                            AssetMaterialization(
-                                asset_key=asset_key,
-                                description=task_run.note,
-                                metadata=task_metadata,
-                            ),
-                        )
-                    )
-                    asset_check_keys_to_request.update(unmigrated_info.checks_per_key[asset_key])
+        current_dag_offset = cursor.dag_query_offset or 0
+        end_date_gte = cursor.end_date_gte or (current_date - timedelta(days=1)).timestamp()
+        end_date_lte = cursor.end_date_lte or current_date.timestamp()
+        sensor_iter = materializations_and_requests_from_batch_iter(
+            end_date_gte=end_date_gte,
+            end_date_lte=end_date_lte,
+            offset=current_dag_offset,
+            airflow_instance=airflow_instance,
+            unmigrated_info=unmigrated_info,
+        )
+        all_materializations: List[Tuple[float, AssetMaterialization]] = []
+        all_check_keys: Set[AssetCheckKey] = set()
+        latest_offset = current_dag_offset
+        while get_current_datetime() - current_date < timedelta(seconds=40):
```

Comment:
40 seems like another number that should be named and is configurable

Reply:
Hmm I don't necessarily think it should be directly configurable, as that would lead to a footgun. I think that this number should probably be whatever the sensor timeout is (which is already configurable) minus 20 seconds.